File: GridView/GridViewIndicatedColumns/GridViewCustomCellsC#/IndicatedColorColumn/IndicatedColorColumn.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Text;
using Telerik.WinControls;
using Telerik.WinControls.UI;

namespace GridViewCustomCells
{ 
   

    public class IndicatedColorColumn : GridViewColorColumn
    {
        private bool enableIndicator;

        public IndicatedColorColumn()
        {
            enableIndicator = true;
        }

        public bool EnableIndicator
        {
            get
            {
                return this.enableIndicator;
            }
            set
            {
                this.enableIndicator = value;
                this.OwnerTemplate.Refresh();
            }
        }

        public override Type GetCellType(GridViewRowInfo row)
        {
            if (row is GridViewDataRowInfo)
            {
                return typeof(IndicatedColorCellElement);
            }
            return base.GetCellType(row);
        }
    }
}

File: GridView/GridViewIndicatedColumns/GridViewCustomCellsVB/IndicatedColorColumn/IndicatedColorColumn.vb
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Text
Imports Telerik.WinControls
Imports Telerik.WinControls.UI

Namespace GridViewCustomCells


	Public Class IndicatedColorColumn
		Inherits GridViewColorColumn


        Private _enableIndicator As Boolean

        Public Sub New()
            _enableIndicator = True
        End Sub

        Public Property EnableIndicator() As Boolean
            Get
                Return Me._enableIndicator
            End Get
            Set(ByVal value As Boolean)
                Me._enableIndicator = value
                Me.OwnerTemplate.Refresh()
            End Set
        End Property

		Public Overrides Function GetCellType(ByVal row As GridViewRowInfo) As Type
			If TypeOf row Is GridViewDataRowInfo Then
				Return GetType(IndicatedColorCellElement)
			End If
			Return MyBase.GetCellType(row)
		End Function
	End Class
End Namespace